### PR TITLE
Add first NXraman Version

### PR DESCRIPTION
**Pull request by RonHildebrandt**
I want to implement the first NeXus definition for NXraman.

As the latest version of NXellipsometry is derived from NXopt, the NXraman here is as well derived from NXopt.
Therefore, the Version here used "NXbeam_path", which will be replaced in the future by using "NXbeam_device" via "previous_devices". This was motivated by the fact, that there is right now no "NXlens_opt(NXbeam_device)" or "NXdetector(NXbeam_device)" [see [Issue #202](https://github.com/FAIRmat-NFDI/nexus_definitions/issues/202)]

**What is necessary?**
From a physical point of view, these are the most important things for a Raman scattering experiment:
1) The incident wavelength (a quite trivial parameter and the spectral shape of a Raman spectrum can be extremely sensitive to this)
2) The scattering configuration, i.e. what is the incident and scattered light direction and polarization.

Beyond this, there are many nice to have parameters, but from my point of view not necessary to be able to use this spectrum.

**Questions:**
A) No symbols were used here, as these were not used in the original NXellipsometry(NXopt) definition. It this fine that way, or at which pointare these symbols used?
B) Is the "scattering_configuration" implemented fine this way? Usually, the "Porto notation" is enough to describe the experiment and beatiful simple. But, it can be the case, that you want to give the specific vectors instead of x,y,z. I just wanted to do this as List of 4 Vectors with dimension 3. Is this okay?
C) Depending on the NXbeam_path or NXbeam_device approach, I may have to completely redo this PR. What is the strategy?

**TL;DR:** Created NXraman.yaml file similar to NXellipsometry.yaml

### CODE REVIEW

File: contributed_definitions/nyaml/NXraman.yaml [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+      doc: |
+        Specify the type of Raman measurement.
+      enumeration: [in situ Raman spectroscopy, resonant Raman spectroscopy, non-resonant Raman spectroscopy, Raman imaging, Tip-enhanced Raman spectroscopy (TERS), Surface-enhanced Raman spectroscopy (SERS), Surface plasmon polariton enhanced Raman scattering (SPPERS), Hyper Raman spectroscopy (HRS), Stimulated Raman spectroscopy (SRS), Inverse Raman spectroscopy (IRS), Coherent anti-Stokes Raman spectroscopy (CARS)]
+    (NXinstrument):
+      doc: |
+        Properties of the Instruments used for Raman instrumeasurements.
+      company:
+        exists: optional
+        doc: |
+          Name of the company which build the instrument.
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:130]: I think we have `NXfabrication` to denote manufacturar information (or NXmanufacturer... I don't fully recall rn).

File: contributed_definitions/nyaml/NXraman.yaml [hunk 2]
```diff
@@ -0,0 +1,755 @@
...
+        However, details of the experiment should be defined in the specific
+        fields of this application definition rather than in this experiment
+        description.
+    experiment_type:
+      doc: |
+        Specify the type of Raman measurement.
+      enumeration: [in situ Raman spectroscopy, resonant Raman spectroscopy, non-resonant Raman spectroscopy, Raman imaging, Tip-enhanced Raman spectroscopy (TERS), Surface-enhanced Raman spectroscopy (SERS), Surface plasmon polariton enhanced Raman scattering (SPPERS), Hyper Raman spectroscopy (HRS), Stimulated Raman spectroscopy (SRS), Inverse Raman spectroscopy (IRS), Coherent anti-Stokes Raman spectroscopy (CARS)]
+    (NXinstrument):
+      doc: |
+        Properties of the Instruments used for Raman instrumeasurements.
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:126]:
```suggestion
        Properties of the instruments used for Raman measurements.
```

Maybe not properties but rather metadata or so?

File: contributed_definitions/nyaml/NXraman.yaml [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
+      doc: |
+        An optional free-text description of the experiment.
+        
+        However, details of the experiment should be defined in the specific
+        fields of this application definition rather than in this experiment
+        description.
+    experiment_type:
+      doc: |
+        Specify the type of Raman measurement.
+      enumeration: [in situ Raman spectroscopy, resonant Raman spectroscopy, non-resonant Raman spectroscopy, Raman imaging, Tip-enhanced Raman spectroscopy (TERS), Surface-enhanced Raman spectroscopy (SERS), Surface plasmon polariton enhanced Raman scattering (SPPERS), Hyper Raman spectroscopy (HRS), Stimulated Raman spectroscopy (SRS), Inverse Raman spectroscopy (IRS), Coherent anti-Stokes Raman spectroscopy (CARS)]
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:123]: Maybe also allow for an other field?

In mpes we came up with a bunch of enumerations, too and it was almost always requested to also have an additional free-text input for everything not fitting to the list.

File: contributed_definitions/nyaml/NXraman.yaml [hunk 4]
```diff
@@ -0,0 +1,755 @@
...
+      company:
+        exists: optional
+        doc: |
+          Name of the company which build the instrument.
+      construction_year(NX_DATE_TIME):
+        exists: optional
+        doc: |
+          ISO8601 date when the instrument was constructed.
+          UTC offset should be specified.
+      software(NXprocess):
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:136]: This should not be an NXprocess

File: contributed_definitions/nyaml/NXraman.yaml [hunk 5]
```diff
@@ -0,0 +1,755 @@
...
+        Specify the type of Raman measurement.
+      enumeration: [in situ Raman spectroscopy, resonant Raman spectroscopy, non-resonant Raman spectroscopy, Raman imaging, Tip-enhanced Raman spectroscopy (TERS), Surface-enhanced Raman spectroscopy (SERS), Surface plasmon polariton enhanced Raman scattering (SPPERS), Hyper Raman spectroscopy (HRS), Stimulated Raman spectroscopy (SRS), Inverse Raman spectroscopy (IRS), Coherent anti-Stokes Raman spectroscopy (CARS)]
+    (NXinstrument):
+      doc: |
+        Properties of the Instruments used for Raman instrumeasurements.
+      company:
+        exists: optional
+        doc: |
+          Name of the company which build the instrument.
+      construction_year(NX_DATE_TIME):
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:131]: This should also belong into NXfabrication

File: contributed_definitions/nyaml/NXraman.yaml [hunk 6]
```diff
@@ -0,0 +1,755 @@
...
+          UTC offset should be specified.
+      software(NXprocess):
+        program:
+          doc: |
+            Commercial or otherwise defined given name of the program that was
+            used to generate the result file(s) with measured data and metadata.
+            This program converts the measured signals to Raman data. If
+            home written, one can provide the actual steps in the NOTE subfield
+            here.
+      incident_source_wavelength(NX_NUMBER):
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:144]:
```suggestion
      incident_wavelength(NX_NUMBER):
```

What is the incident _source_? What other incident wavelengths are there?

Comment (domna) [reply contributed_definitions/nyaml/NXraman.yaml:144]: Why is this not part of the NXsource or NXbeam group?

File: contributed_definitions/nyaml/NXraman.yaml [hunk 7]
```diff
@@ -0,0 +1,755 @@
...
+          (i.e for white light laser source)
+      scattering_configuration(NX_CHAR):
+        # unit: NX_LENGTH
+        doc: |
+          Scattering configuration as defined by the porto notation by three
+          states, which are othogonal to each other. Example: z(xx)z for
+          parallel polarized backscattering configuration.
+          
+          See:
+          https://www.cryst.ehu.es/cgi-bin/cryst/programs/nph-doc-raman
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:159]: Maybe a proper reference is better? Nobody guarantees that this page will stay as it is and be available in 10 years.

File: contributed_definitions/nyaml/NXraman.yaml [hunk 8]
```diff
@@ -0,0 +1,755 @@
...
+          Which indicent wavelength was used for e.g. a laser source.
+          For multiple excitation wavelengths, please state the specific discret
+          wavelengths (i.e. for dye laser) or the specific contnious range
+          (i.e for white light laser source)
+      scattering_configuration(NX_CHAR):
+        # unit: NX_LENGTH
+        doc: |
+          Scattering configuration as defined by the porto notation by three
+          states, which are othogonal to each other. Example: z(xx)z for
+          parallel polarized backscattering configuration.
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:156]: Maybe a bit of introduction on the porto notation here? Not everyone might be used to this I think

File: contributed_definitions/nyaml/NXraman.yaml [hunk 9]
```diff
@@ -0,0 +1,755 @@
...
+            doc: |
+              3 x 4 Matrix, which lists the porto notation vectors A, B, C, D.
+              A has to be perpendicular to B and C perpendicular to D.
+            dim: [[1, 4], [2, 3]]
+      opt_source(NXbeam_device):
+        doc: |
+          beam device instance of the optical source.
+        device:
+          doc: |
+            It should contain the name of the Nexus source group.
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:190]: The docstring read a bit as work in progress?

File: contributed_definitions/nyaml/NXraman.yaml [hunk 10]
```diff
@@ -0,0 +1,755 @@
...
+          Dispersive element to create e.g. an energetic resolution used in the 
+          measurement.
+        device:
+          doc: |
+            It should contain the name of the Nexus Monochromator group
+      (NXmonochromator):
+        doc: |
+          Monochromator with arbitrary choice of the name, which shall be referenced
+          by the spectrometer(NXbeam_device).
+      Sample_as_beam_element(NXbeam_device):
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:316]:
```suggestion
      sample_as_beam_element(NXbeam_device):
```

I think it's a typo or is it meant to also contain names like `bananample_as_beam_element`?

File: contributed_definitions/nyaml/NXraman.yaml [hunk 11]
```diff
@@ -0,0 +1,755 @@
...
+        doc: |
+          Enables beam path description of the sample in the setup.
+          it in a beam path.
+        device:
+          doc: |
+            Contains the link to the sample.
+    (NXsample):
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:323]: You might want to require a name or molecular formula of the sample or at least recommend them

File: contributed_definitions/nyaml/NXraman.yaml [hunk 12]
```diff
@@ -0,0 +1,755 @@
...
+    data_collection(NXprocess):
+      data_type:
+        doc: |
+          Select which type of data was recorded, for example a simple spectrum
+          with intensity vs. Raman shift, CCD image or a set of spectra with
+          probe, reference and background signals for pump beam on and off.
+        enumeration: [Integrated single spectrum, CCD image, Probe + Background + Reference for Pump-on and Pump-off (i.e. for FSRS)]
```
Comment (domna) [inline contributed_definitions/nyaml/NXraman.yaml:330]: I don't understand this part. `NXprocess` is intended to describe post-processing of data. Here it's just used to denote which typo of data was collected. For me this rather belongs into the (NXdetector) group or so

File: contributed_definitions/nyaml/NXbeam_device.yaml [hunk 13]
```diff
@@ -31,6 +34,10 @@ NXbeam_device(NXobject):
       all these devices have the group name "second harmonic generation".
       Is used for simplified setup vizualization (or description?).
 
+  device:
+    doc: |
+      This is the name of the NeXus device like a NXdetector, a NXmonochromator or NXsource.
```
Comment (domna) [inline contributed_definitions/nyaml/NXbeam_device.yaml:39]: Consider making this a link, i.e., a reference to a full path of data entry. Something like `/entry/instrument/monochromator`. Then it's clear to which detector this refers,

see [here](https://github.com/FAIRmat-NFDI/nexus_definitions/blob/da544c9be46a0d77d877915f59b3a13b698c0c8f/contributed_definitions/nyaml/NXmpes.yaml#L745) for an example

**Review by domna — COMMENTED** [review]
While going through this I thought it might be helpful to group the beam_devices together in one sub-group. Then it might be easier to understand how they belong together.

File: contributed_definitions/nyaml/NXraman.yaml (hunk 1 above)
Comment (RonHildebrandt) [reply contributed_definitions/nyaml/NXraman.yaml:130]: sounds good. I've replaced it with that.

File: contributed_definitions/nyaml/NXraman.yaml (hunk 3 above)
Comment (RonHildebrandt) [reply contributed_definitions/nyaml/NXraman.yaml:123]: All right, ive added this with addition to specify the option "other" by another field

File: contributed_definitions/nyaml/NXraman.yaml (hunk 4 above)
Comment (RonHildebrandt) [reply contributed_definitions/nyaml/NXraman.yaml:136]: Ahhh okay :D I just copied it from NXellipsometry. Thaught this was okay. In principle up to now, it should be a free text to enter some information?

File: contributed_definitions/nyaml/NXraman.yaml (hunk 10 above)
Comment (RonHildebrandt) [reply contributed_definitions/nyaml/NXraman.yaml:316]: yea - this sounds good. Maybe in a future project we will need this :x (yea its a typo)

File: contributed_definitions/nyaml/NXraman.yaml (hunk 11 above)
Comment (RonHildebrandt) [reply contributed_definitions/nyaml/NXraman.yaml:323]: ah okay. Ive added some exists: with required and recommended.

File: contributed_definitions/nyaml/NXraman.yaml (hunk 12 above)
Comment (RonHildebrandt) [reply contributed_definitions/nyaml/NXraman.yaml:330]: This is defined in this way in NXopt. Maybe thats a remaining thing from the Harmonization of NXellipsometry and NXopt?
Right, from the definition of NXprocess, this does not seem to be proper.
No action done yet - have to discuss this.

File: contributed_definitions/nyaml/NXraman.yaml (hunk 2 above)
Comment (RonHildebrandt) [reply contributed_definitions/nyaml/NXraman.yaml:126]: Ive addressed this

**Comment by RonHildebrandt** [comment]
I like to close this PR, as this will be implemented in PR220 (https://github.com/FAIRmat-NFDI/nexus_definitions/pull/220)
